### PR TITLE
Fix double tunsetif and invalid cstring handling on linux

### PR DESCRIPTION
**Pull request by ThrasherLT**
`new_from_tun` function contained 2 bugs:
- For interface name `.len()` would return a value that's equal to the buffer size and later treated as invalid by Telio, because it included multiple trailing nul bytes, since the name wasn't properly handled as a CString.
- Leftover code caused IOCtl with TUNSETIF to be called a second time on an already confiugred tunnel, even though it's already been called with TUNGETIF a few lines before.

So the solutions were:
- Implement proper CString handling.
- Remove the extra TUNSETIF call.

### CODE REVIEW

File: neptun/src/device/tun_linux.rs [hunk 1]
```diff
@@ -140,29 +140,11 @@ impl TunSocket {
...
+        let name = std::ffi::CStr::from_bytes_until_nul(&ifr.ifr_name)
+            .map_err(|_| Error::InvalidTunnelName)?
+            .to_str()
             .map_err(|_| Error::InvalidTunnelName)?
             .to_owned();
-
-        #[cfg(target_os = "linux")]
-        {
-            ifr.ifr_ifru = IfrIfru {
-                ifru_flags: (IFF_TUN | IFF_MULTI_QUEUE) as _,
```
[the remark this answers is not in the document]
Comment (tomasz-grz) [reply neptun/src/device/tun_linux.rs:150]: Ahh I see, or is this the duplicate from [new()](https://github.com/NordSecurity/NepTUN/blob/ee37daadeb0dc6f8e6125f7a51c2dd0a47fc73bd/neptun/src/device/tun_linux.rs#L93)

**Comment by Hasan6979** [comment]
+1

File: neptun/src/device/tun_linux.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ThrasherLT) [reply neptun/src/device/tun_linux.rs:150]: Yeah, I think it's a duplicate from `new()`, as in the description. Essentially, you can't call  `IOCtl` with `TUNGETIF` if you haven't called it with `TUNSETIF` before, but you can only call it with `TUNSETIF` once, so this code doesn't really make sense. These flags are expected to be set before passing the tunnel to Libtelio.

**Review by tomasz-grz — APPROVED** [review]
+1

**Pull request merged**